easybuild/easyblocks/i/icc.py

```diff
@@ -105,53 +117,117 @@ def sanity_check_step(self):
     def make_module_req_guess(self):
         """Customize paths to check and add in environment.
         """
+        debuggerpath = None
+        prefix = None
         if self.cfg['m32']:
             # 32-bit toolchain
+            libpaths = ['lib', 'lib/ia32'],
             dirmap = {
                 'PATH': ['bin', 'bin/ia32', 'tbb/bin/ia32'],
-                'LD_LIBRARY_PATH': ['lib', 'lib/ia32'],
-                'LIBRARY_PATH': ['lib', 'lib/ia32'],
+                'LD_LIBRARY_PATH': libpaths,
+                'LIBRARY_PATH': libpaths,
                 'MANPATH': ['man', 'share/man', 'man/en_US'],
                 'IDB_HOME': ['bin/intel64']
             }
         else:
-            # 64-bit toolit
-            dirmap = {
-                'PATH': ['bin', 'bin/intel64', 'tbb/bin/emt64'],
-                'LD_LIBRARY_PATH': ['lib', 'lib/intel64'],
-                'LIBRARY_PATH': ['lib', 'lib/intel64'],
-                'MANPATH': ['man', 'share/man', 'man/en_US'],
-                'IDB_HOME': ['bin/intel64']
-            }
+            # 64-bit toolkit
+
+            # using get_software_version('GCC') won't work, while the compiler toolchain is dummy:dummy, which does not
+            # load dependencies.
+            gccversion = get_software_version('GCC')
+            # manual approach to at least have the system version of gcc
+            if not gccversion:
+                cmd = 'gcc --version'
+                (out, _) = run_cmd(cmd, log_all=True, simple=False)
+                ver_re = re.compile("^gcc \(GCC\) (?P<version>[0-9.]+) [0-9]+", re.M)
+                gccversion = ver_re.search(out).group('version')
+                self.log.warning("Using system GCC version %s" % gccversion)
+
+            # TBB directory structure
+            # https://www.threadingbuildingblocks.org/docs/help/tbb_userguide/Linux_OS.htm
+            tbbgccversion = 'gcc4.4'  # gcc version 4.4 or higher that may or may not support exception_ptr
+            if gccversion and LooseVersion(gccversion) >= LooseVersion("4.1") and LooseVersion(gccversion) < LooseVersion("4.4"):
+                tbbgccversion = 'gcc4.1'  # gcc version number between 4.1 and 4.4 that do not support exception_ptr
```

Comment:
please flesh out this whole block into a separate function/method (maybe even in the TBB easyblock, where it belongs)?